rules/internal/utils/casing.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+// ToUpperCamelCase returns the UpperCamelCase of a string, including removing
+// delimiters (_,-,., ) and using them to denote a new word.
+func ToUpperCamelCase(s string) string {
```

Comment:
Sooooo we could totally switch to using this but....we already use `strcase` for this type of stuff: https://pkg.go.dev/github.com/stoewer/go-strcase 😬 Not sure we need this file, unless we want to drop the external/non-google dependency (which would be fine if that was a goal, I do see value in such a "clean" module approach)

Reply:
strcase doesn't filter out "-" or "_" unfortunately. I did see some hacks with double-conversion to snake case, then to upperCamelCase, but I figured it might be best to roll our own that matches our own definition of "upperCamelCase".

I did find this version of strcase that matches our definition: https://github.com/iancoleman/strcase. But I decided against switching it because we need to vet that it works for the other use cases as well.

Reply:
ack, here's a tracking issue: https://github.com/googleapis/api-linter/issues/1150. Agreed that mixed usage is confusing: i'll follow up to try to clean that up. 

Once nice thing about our own handrolled one is we can add options as needed.